src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1166,6 +1171,31 @@ private BoundExpression BindDefaultExpression(DefaultExpressionSyntax node, Diag
             return expression;
         }
 
+        /// <summary>
+        /// Is this is an _ identifier in a context where discards are allowed?
+        /// </summary>
+        private bool FallBackOnDiscard(SimpleNameSyntax node, DiagnosticBag diagnostics)
+        {
+            if (node.Identifier.ContextualKind() != SyntaxKind.UnderscoreToken)
+            {
+                return false;
+            }
+
+            CSharpSyntaxNode containingDeconstruction = node.GetContainingDeconstruction();
```

Comment:
`GetContainingDeconstruction` and `IsOutVarDiscardIdentifier` behave correctly if they encounter GenericNameSyntax, but I added a check for `IdentifierNameSyntax` above for clarity, and a test.

---
In reply to: [88535608](https://github.com/dotnet/roslyn/pull/15192#discussion_r88535608) [](ancestors = 88535608)